pkg/codec/modifier_base.go

```diff
@@ -242,16 +252,27 @@ func doForMapElements[T any](valueMapping map[string]any, fields map[string]T, f
 		path := strings.Split(key, ".")
 		name := path[len(path)-1]
 		path = path[:len(path)-1]
+
 		extractMaps, err := getMapsFromPath(valueMapping, path)
 		if err != nil {
-			return err
+			return PathMappingError{Err: err, Path: key}
 		}
 
 		for _, em := range extractMaps {
 			if err = fn(em, name, value); err != nil {
-				return err
+				return PathMappingError{Err: err, Path: key}
 			}
 		}
 	}
+
 	return nil
 }
+
+type PathMappingError struct {
+	Err  error
+	Path string
+}
+
+func (e PathMappingError) Error() string {
+	return fmt.Sprintf("mapping error for path (%s): %s", e.Path, e.Err.Error())
```

Comment:
`Error()` should be called automatically for `error`s:
```suggestion
	return fmt.Sprintf("mapping error for path (%s): %s", e.Path, e.Err)
```
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

https://pkg.go.dev/fmt

Reply:
Haha. This is just a pattern I have gotten into due to a linter I have activated elsewhere. ¯\_(ツ)_/¯